src/test/java/com/internship/socialnetwork/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,177 @@
+
+        UserDto userDto = userService.getUser(userId1);
+
+        assertNotNull(userDto);
+        assertEquals(username1, userDto.getUsername());
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void shouldThrowNotFoundException_whenGetUser_ifUserDoesNotExist() {
+        when(userRepository.findById(userId2)).thenReturn(Optional.empty());
+
+        userService.getUser(userId2);
+    }
+
+    @Test
+    public void shouldGetAllUsers() {
+        List<User> users = new ArrayList<>();
+
+        users.add(User.builder().username(username1).build());
+        users.add(User.builder().username(username2).build());
+
+        when(userRepository.findAll()).thenReturn(users);
+
+        List<UserDto> userDtos = userService.getAllUsers();
+
+        assertEquals(2, userDtos.size());
+    }
+
+    @Test
+    public void testCreateUser_Success() {
+        UserDto userDto = UserDto.builder()
+                .username(username1)
+                .name(name1)
+                .surname(surname1)
+                .email(email)
+                .build();
+
+        when(userRepository.findByUsernameOrEmail(anyString(), anyString())).thenReturn(Optional.empty());
+
+        User savedUser = User.builder()
+                .username(username1)
+                .name(name1)
+                .surname(surname1)
+                .email(email)
+                .build();
+
+        when(userRepository.save(any(User.class))).thenReturn(savedUser);
+
+        UserDto createdUserDto = userService.createUser(userDto);
+
+        assertEquals(userDto.getUsername(), createdUserDto.getUsername());
+        assertEquals(userDto.getName(), createdUserDto.getName());
+        assertEquals(userDto.getSurname(), createdUserDto.getSurname());
+        assertEquals(userDto.getEmail(), createdUserDto.getEmail());
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void shouldThrowBadRequestException_whenUserAlreadyExists() {
+        UserDto existingUserDto = UserDto.builder()
+                .username(username1)
+                .name(name1)
+                .surname(surname1)
+                .email(email)
+                .build();
+
+        when(userRepository.findByUsernameOrEmail(anyString(), anyString())).thenReturn(Optional.of(existingUserDto.covertToModel()));
+
+        userService.createUser(existingUserDto);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointerException_whenUsernameIsNull() {
+        UserDto userDto = new UserDto();
+        userDto.setUsername(null);
+
+        userService.createUser(userDto);
+    }
+
+    @Test
+    public void shouldGetUserModel() {
+        User user = new User();
+        user.setId(userId1);
+        user.setUsername(username1);
+        when(userRepository.findById(userId1)).thenReturn(Optional.of(user));
+        User user1 = userService.getUserModel(userId1);
+        assertNotNull(user1);
```

Comment:
Pls divide setup, method call and asserts with a blank line and comments mentioned above